lib/ansible/plugins/action/unarchive.py

```diff
@@ -38,9 +38,20 @@ def run(self, tmp=None, task_vars=None):
 
         source  = self._task.args.get('src', None)
         dest    = self._task.args.get('dest', None)
-        copy    = boolean(self._task.args.get('copy', True))
+        remote_src = boolean(self._task.args.get('remote_src', False))
         creates = self._task.args.get('creates', None)
 
+        # "copy" is deprecated in favor of "remote_src".
+        if 'copy' in self._task.args:
+            # They are mutually exclusive.
+            if 'remote_src' in self._task.args:
+                result['failed'] = True
+                result['msg'] = "parameters are mutually exclusive: ('copy', 'remote_src')"
+                return result
+            # We will take the information from copy and store it in
+            # the remote_src var to use later in the playbook
```

Comment:
Hmmm, I don't think they are mutually exclusive. One deprecates the other and the error message should reflect that IMO. Anyhow, fine by me.